packages/api/amplify_api/lib/src/graphql/graphql_send_request.dart

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:convert';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+
+import '../util.dart';
+import 'graphql_response_decoder.dart';
+
+@internal
+Future<GraphQLResponse<T>> graphQLSendRequest<T>({
+  required GraphQLRequest<T> request,
+  required http.Client client,
+  required Uri uri,
+}) async {
+  final body = {'variables': request.variables, 'query': request.document};
+  final graphQLResponse = await client.post(uri, body: json.encode(body));
+
+  final responseBody = json.decode(graphQLResponse.body);
+
+  if (responseBody is! Map<String, dynamic>) {
+    throw Exception('Response from GraphQL client incorrect type.');
+  }
+
+  // Handle Model responses that do not return a 'data' key
+  final responseData =
+      request.decodePath != null ? responseBody : responseBody['data'];
```

Comment:
Is this ternary necessary here? I ran the integ tests and saw lots of errors with the model helpers. I was thinking you would always want `responseBody['data']` because model helpers won't change the server response structure, just changes how the client decodes the response and doesn't require a change here. I think it was working before, just was suggesting adding some tests for it.

Reply:
Resolved offline. Ternary is not needed.